code/game/machinery/droneDispenser.dm

```diff
@@ -57,38 +58,61 @@
 	materials.insert_amount_mat(starting_amount)
 	materials.precise_insertion = TRUE
 	using_materials = list(/datum/material/iron = iron_cost, /datum/material/glass = glass_cost, /datum/material/copper = copper_cost, /datum/material/gold = gold_cost)
+	if(has_parts)
```

Comment:
o co tu chodzi? czemu w pierwszej wersji nie ma iron/twenty?

Reply:
tak w sumie to po co te drone dispensery bez części?

Reply:
jest kilka wariantów dyspenserów - część z nich to np. Fabrykator Hivebotów - nie chcę, żeby miał stockowe części bo raz, ma fixed czas produkcji (1s), dwa, tym bardziej nie chcę, żeby jakiś spessmen wpadł na pomysł, żeby ulepszyć ten fabrykator częściami T4 bo wtedy dzieje się ujemny czas produkcji a to złe juju.

W związku z tym jest zmienna `has_parts` - jeżeli chcę, żeby dyspenser dało się dekonstruować i ulepszać - ma części i płytkę. W przypadku podtypów które nie chcę, żeby gracze mogli ulepszać czy dekonstruować - dropi przy zniszczeniu/dekonstrukcji 20 metalu zamiast tego.

Reply:
wydaje mi sie ze lepiej by to bylo podzielic na 2 podtypy ale niech już będzie